src/overlays/actors/ovl_En_Bili/z_en_bili.c

```diff
@@ -46,69 +59,705 @@ static ColliderCylinderInit D_809C1640 = {
+    s32 i;
+
+    if (this->actor.draw != NULL) {
+        if (this->actor.flags & 0x8000) {
+            return;
+        }
+        this->actor.draw = NULL;
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0x50);
+    }
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer != 0) {
+        for (i = 0; i < 2; i++) {
+            effectPos.x = ((Rand_ZeroOne() * 10.0f) + this->actor.world.pos.x) - 5.0f;
+            effectPos.y = ((Rand_ZeroOne() * 5.0f) + this->actor.world.pos.y) - 2.5f;
+            effectPos.z = ((Rand_ZeroOne() * 10.0f) + this->actor.world.pos.z) - 5.0f;
+
+            effectVelocity.y = Rand_ZeroOne() + 1.0f;
+            effectScale = Rand_S16Offset(40, 40);
+
+            if (Rand_ZeroOne() < 0.7f) {
+                EffectSsDtBubble_SpawnColorProfile(globalCtx, &effectPos, &effectVelocity, &effectAccel, effectScale,
+                                                   25, 2, 1);
+            } else {
+                EffectSsDtBubble_SpawnColorProfile(globalCtx, &effectPos, &effectVelocity, &effectAccel, effectScale,
+                                                   25, 0, 1);
+            }
+        }
+    } else {
+        Actor_Kill(&this->actor);
+    }
+
+    if (this->timer == 14) {
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 40, NA_SE_EN_BIRI_BUBLE);
+    }
+}
+
+void EnBili_Stunned(EnBili* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->actor.bgCheckFlags & 2) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+    }
+    if (this->timer == 0) {
+        EnBili_SetupFloatIdle(this);
+    }
+}
+
+void EnBili_Frozen(EnBili* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (!(this->actor.flags & 0x8000)) {
+        this->actor.gravity = -1.0f;
+    }
+
+    if ((this->actor.bgCheckFlags & 1) || (this->actor.floorHeight == -32000.0f)) {
+        this->actor.colorFilterTimer = 0;
+        EnBili_SetupDie(this);
+    } else {
+        this->actor.colorFilterTimer = 10;
+    }
+}
+
+void EnBili_UpdateDamage(EnBili* this, GlobalContext* globalCtx) {
+    u8 damageEffect;
+
+    if ((this->actor.colChkInfo.health != 0) && (this->collider.base.acFlags & AC_HIT)) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        func_80035650(&this->actor, &this->collider.info, 1);
+        if ((this->actor.colChkInfo.damageEffect != 0) || (this->actor.colChkInfo.damage != 0)) {
+            if (Actor_ApplyDamage(&this->actor) == 0) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_BIRI_DEAD);
+                func_80032C7C(globalCtx, &this->actor);
+                this->actor.flags &= ~1;
+            }
+
+            damageEffect = this->actor.colChkInfo.damageEffect;
+
+            if (damageEffect == 0x1) { // Deku Nut
+                if (this->actionFunc != EnBili_Stunned) {
```

Comment:
you can make an enum for the damage effect if you like. you would use them both in the damage table and in the code